API.md

```diff
@@ -437,6 +438,27 @@ append(4, [1, 2, 3]); // Iterable[1, 2, 3, 4]
 
 See [append](#append)
 
+### distinct
+
+**distinct(selector, [source](#wrappable))**  
+**distinct([source](#wrappable))**  
+**__distinct([source](#iterable), ?selector)**  
+
+Yields only the first occurrance of each value from `source`. An optional `selector` can be used to specify a transform function which returns a key to distinguish the values.
+
+```js
+distinct([3, 1, 4, 1, 5, 9, 2, 6, 5, 3]); // Iterable[3, 1, 4, 5, 9, 2, 6]
+distinct((x) => x % 7, [14, 21, 9, 777]); // Iterable[0, 2]
```

Comment:
It does look like the implementation is correct, it's just the example output that's wrong.

Reply:
> distinct((x) => x % 7, [14, 21, 9, 777]); // Iterable[14, 9]

Nah, all of 14, 21 and 777 has a modulo 7 result of 0. Modulo gives you the remainder after dividing.

I'll try to find simpler yet still meaningful examples.

Reply:
Right. Meaning that of those numbers only 14 should appear in the output, because the other numbers are considered identical to it. 9 also makes it in because it's the first value for which the `x % 7 === 2`.

Reply:
The output can only contain values that were in the input (not 0 or 2).

Reply:
Oh, you are definitely right.

Reply:
I actually think a simpler version of that modulo example might be an excellent one to use since it demonstrates that property which you've just proven may seem surprising.

Reply:
fixed in 45b72934c52c0df4e907d05d07b1429a8bc7dc43